### PR TITLE
Fix cosmos query metrics no longer logged

### PR DESCRIPTION
**Pull request by Netyyyy**
As title.
Fixes https://github.com/Azure/azure-sdk-for-java/issues/31238

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/cosmos/properties/AzureCosmosProperties.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 public class AzureCosmosProperties extends AbstractAzureServiceConfigurationProperties implements CosmosClientProperties {
 
     public static final String PREFIX = "spring.cloud.azure.cosmos";
+    public static final String POPULATE_QUERY_METRICS = "spring.cloud.azure.cosmos.populate-query-metrics";
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/cosmos/properties/AzureCosmosProperties.java:25]: Why adding this?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-actuator-autoconfigure/pom.xml [hunk 2]
```diff
@@ -66,6 +66,12 @@
       <version>4.38.0</version> <!-- {x-version-update;com.azure:azure-cosmos;dependency} -->
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-spring-data-cosmos</artifactId>
+      <version>3.29.0</version> <!-- {x-version-update;com.azure:azure-spring-data-cosmos;dependency} -->
+      <optional>true</optional>
+    </dependency>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-actuator-autoconfigure/pom.xml:74]: And if you move the bean configuration to `CosmosDataAutoConfiguration`, do we still need this?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            this.contextRunner
+                .withBean(CosmosClientBuilder.class, () -> cosmosClientBuilder)
+                .withBean(CosmosClient.class, () -> mock(CosmosClient.class))
+                .withPropertyValues(
+                    "spring.cloud.azure.cosmos.endpoint=" + ENDPOINT,
+                    "spring.cloud.azure.cosmos.database=test-database",
+                    "spring.cloud.azure.cosmos.populate-query-metrics=false")
+                .run(context -> assertThat(context).doesNotHaveBean(ResponseDiagnosticsProcessor.class));
+        }
+    }
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java:69]: Can we have an assertion that without the ResponseDiagnosticsProcessor bean, spring data cosmos can be configured correctly?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        this.contextRunner
+            .withConfiguration(AutoConfigurations.of(ResponseDiagnosticsProcessorConfiguration.class))
+            .run(context -> {
+                ResponseDiagnosticsProcessor processor = (ResponseDiagnosticsProcessor) context.getBean("ResponseDiagnosticsProcessor");
+                assertTrue(processor instanceof ResponseDiagnosticsProcessorExtend);
+            });
+    }
+
+    @Configuration
+    @AutoConfigureBefore(CosmosDataDiagnosticsConfiguration.class)
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java:82]: No need of this.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                    "spring.cloud.azure.cosmos.database=test-database",
+                    "spring.cloud.azure.cosmos.populate-query-metrics=false")
+                .run(context -> assertThat(context).doesNotHaveBean(ResponseDiagnosticsProcessor.class));
+        }
+    }
+
+    @Test
+    void configureWithUserProvideResponseDiagnosticsProcessor() {
+        this.contextRunner
+            .withConfiguration(AutoConfigurations.of(ResponseDiagnosticsProcessorConfiguration.class))
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java:74]: withUserConfiguration instead, the ResponseDiagnosticsProcessorConfiguration should not a AutoConfiguration class, it's provided by the user.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+class CosmosDataDiagnosticsConfigurationTests {
+
+    private static final String ENDPOINT = "https://test.documents.azure.com:443/";
+    private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+        .withConfiguration(AutoConfigurations.of(
+            AzureGlobalPropertiesAutoConfiguration.class,
+            AzureCosmosAutoConfiguration.class,
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java:31]: Do we need this?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            CosmosDataAutoConfiguration.class));
+    CosmosClientBuilder cosmosClientBuilder = mock(CosmosClientBuilder.class);
+    CosmosAsyncClient cosmosAsyncClient = mock(CosmosAsyncClient.class);
+
+    @Test
+    void configureWithPopulateQueryMetricsEnabled() {
+        try (MockedStatic<CosmosFactory> mockedStatic = mockStatic(CosmosFactory.class, RETURNS_MOCKS)) {
+            when(cosmosClientBuilder.buildAsyncClient()).thenReturn(cosmosAsyncClient);
+            mockedStatic.when(() -> CosmosFactory.createCosmosAsyncClient(cosmosClientBuilder))
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java:40]: I think we don't need to mock cosmos factory?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataAutoConfigurationTests.java [hunk 8]
```diff
@@ -50,9 +50,9 @@ void configureWithoutEndpoint() {
     }
 
     @Test
+    @SuppressWarnings("try")
```
[the remark this answers is not in the document]
Comment (Netyyyy) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataAutoConfigurationTests.java:52]: https://dev.azure.com/azure-sdk/public/_build/results?buildId=1932594&view=logs&j=6438ed74-af23-540e-94f3-38d1eefc9e9b&t=61f12b16-07fc-5eac-62aa-0de5efacf35e&l=33
image

**Review by saragluna — APPROVED** [review]

**Pull request merged**